### PR TITLE
Updated docs for HTMLElement.dataset property checks

### LINKED ISSUE #28693: HTMLElement.dataset: using the `in` operator is unsafe
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

### What specific section or headline is this issue about?

Accessing Values

### What information was incorrect, unhelpful, or incomplete?

The `in` operator is exposed to the usual risk of prototype pollution, so anything added to Object.prototype and other classes in the inheritance chain of `DOMStringMap` will be `in` the dataset but not in `hasOwnProperty` .

### What did you expect to see?

Change `in` to `hasOwnProperty` or add a caveat.

### Do you have any supporting links, references, or citations?

```javascript
	Object.prototype.bad = function() { console.log("bad prop")};
	var e = document.body.querySelector("#test"); // <span id="test" data-man="1">
	console.log("man", "man" in e.dataset, "bad", "bad" in e.dataset); // man true, bad true
	console.log("man", e.dataset.hasOwnProperty("man"), 
            "bad", e.dataset.hasOwnProperty("bad"));// man true, bad false

```

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/htmlelement/dataset`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/htmlelement/dataset/index.md
* Last commit: https://github.com/mdn/content/commit/a3d9f61a8990ba7b53bda9748d1f26a9e9810b18
* Document last modified: 2023-04-07T00:39:15.000Z

</details>

### PR DESCRIPTION
**Pull request by kevinctofel**
### Description

Adding hasOwn() example to the section for accessing element properties.

### Motivation

An issue was opened to request this addition because current documentation could expose web apps to property pollution. This change will inform readers that there are two ways to access dataset properties and one will mitigate against property pollution.

### Additional details

- Discussion in the issue thread suggested this document addition to supplement current docs.
- Note: Only my second PR, so apologies in advance if there are any inadvertent errors. Happy to address them, of course. 

### Related issues and pull requests

#28693

### CODE REVIEW

File: files/en-us/web/api/htmlelement/dataset/index.md [hunk 1]
```diff
@@ -115,6 +120,12 @@ if (!("someDataAttr" in el.dataset)) {
   // Result on JS: 'someDataAttr' in el.dataset === true
   // Result on HTML: <div id="user" data-id="1234567890" data-user="carinaanand" data-some-data-attr="mydata">Carina Anand</div>
 }
+
+if (!(el.dataset.hasOwn("someDataAttr"))) {
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlelement/dataset/index.md:124]: This is not how you use `hasOwn`. Please double-check the reference.

File: files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Safari and Firefox on desktop do not support the Badging API and do not support
 
 ### Mobile support
 
-On mobile operating systems, badges are supported on Chromium-based browsers running on Android.
+Badges are supported on mobile operating systems, including Chromium-based browsers running on Android and in Safari on iOS and iPadOS, starting with iPadOS 16.4.
```
Comment (Josh-Cena) [inline files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:32]: Changes in this file doesn't seem related. You probably forgot to start a new branch from main?

Comment (kevinctofel) [reply files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:32]: Ugh, you are correct. There are files from a previous PR I worked on. Learning but slowly! 

Thinking it's best to somehow remove this PR and properly fork/branch from main to address this and the hasOwn() usage, yes? Sorry about this!

Comment (Josh-Cena) [reply files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:32]: Exactly, you should create a branch for every PR and leave main purely for pulling our changes. Right now you may want to delete the fork and redo things.

Comment (kevinctofel) [reply files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:32]: Will do. Apologies again for the inexperience and for any extra work I might have caused.

Comment (Josh-Cena) [reply files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:32]: It's totally okay. One of our goals as an educational organization is to get you prepared to use GitHub confidently :)

**Comment by Josh-Cena** [comment]
Going to close this and you should start a new PR instead
